### ISSUE TITLE
`SyntaxError` caused by more-itertools in example-test

### ISSUE THREAD

**Issue by niboshi** [chainer/chainer#7937]
https://jenkins.preferred.jp/job/chainer/job/cupy_pr/274/TEST=cupy-example,label=mn1-p100/console

```
14:25:33 + coverage run -a --branch examples/kmeans/kmeans.py -m 1
14:25:34 Traceback (most recent call last):
14:25:34   File "examples/kmeans/kmeans.py", line 9, in <module>
14:25:34     import cupy
14:25:34   File "/home/user/.local/lib/python2.7/site-packages/cupy/__init__.py", line 59, in <module>
14:25:34     from cupy import testing  # NOQA  # NOQA
14:25:34   File "/home/user/.local/lib/python2.7/site-packages/cupy/testing/__init__.py", line 2, in <module>
14:25:34     from cupy.testing import attr  # NOQA
14:25:34   File "/home/user/.local/lib/python2.7/site-packages/cupy/testing/attr.py", line 6, in <module>
14:25:34     import pytest
14:25:34   File "/usr/lib/python2.7/site-packages/pytest.py", line 14, in <module>
14:25:34     from _pytest.fixtures import fillfixtures as _fillfuncargs
14:25:34   File "/usr/lib/python2.7/site-packages/_pytest/fixtures.py", line 16, in <module>
14:25:34     from more_itertools import flatten
14:25:34   File "/usr/lib/python2.7/site-packages/more_itertools/__init__.py", line 1, in <module>
14:25:34     from more_itertools.more import *  # noqa
14:25:34   File "/usr/lib/python2.7/site-packages/more_itertools/more.py", line 340
14:25:34     def _collate(*iterables, key=lambda a: a, reverse=False):
14:25:34                                ^
14:25:34 SyntaxError: invalid syntax
```

**Comment by niboshi** [comment]
The last version of [more-itertools](https://pypi.org/project/more-itertools/) that supports Python 2 was more-itertools==5.0.0.

On the other hand, [the code in question](https://github.com/erikrose/more-itertools/blob/6.0.0/more_itertools/more.py#L329) was only merged in 6.0.0.

What does that mean 🤔

**Comment by kmaehashi** [comment]
This issue has been resolved in https://github.com/chainer/chainer-test/pull/521.
`pip` used in the Py2-on-CentOS test was too old so it cannot detect more_itertools 6.0.0+ does ot support Py2.
